### PR TITLE
Allow other engine builders other than docker

### PR DESCRIPTION
**Pull request by spolti**
chore:	This change will allow user to use other engine builds.
	In addition, for Podman/Buildah engine builder, users can
	set the target architecture, useful when using ARM and
	there is a need to build x86 Container images.
	Can be done with `ENGINE=podman ARCH="--arch x86_64" make docker...`

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

**Comment by spolti** [comment]
cc @sivanantha321 @yuzisun

**Review by sivanantha321 — DISMISSED** [review]

**Comment by spolti** [comment]
@yuzisun @terrytangyuan do you think this can be merged?

File: Makefile [hunk 1]
```diff
@@ -196,8 +202,8 @@ bump-version:
 	echo "bumping version numbers for this release"
 
 # Build the docker image
-docker-build: test
-	docker buildx build . -t ${IMG}
+docker-build: #test
```
Comment (terrytangyuan) [inline Makefile]: Is this change intended?

Comment (spolti) [reply Makefile]: no :)
thanks

**Review by terrytangyuan — COMMENTED** [review]
How about "DOCKER_BINARY" or "CONTAINER_ENGINE"?

**Comment by spolti** [comment]
IMHO _CONTAINER_ENGINE_ is better than _DOCKER_BINARY_.

Also, we have the option to use _BUILDER_ENGINE_.

**Comment by spolti** [comment]
@terrytangyuan @yuzisun hey, can you please take a look?

**Review by terrytangyuan — APPROVED** [review]
/lgtm
/approve
